internal/cmd/kafka/command_consumergroup_list_onprem.go

```diff
@@ -0,0 +1,54 @@
+package kafka
+
+import (
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/kafkarest"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+func (c *consumerGroupCommand) newListCommandOnPrem() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List Kafka consumer groups.",
+		Args:  cobra.NoArgs,
+		RunE:  c.listOnPrem,
+	}
+
+	cmd.Flags().AddFlagSet(pcmd.OnPremKafkaRestSet())
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *consumerGroupCommand) listOnPrem(cmd *cobra.Command, args []string) error {
+	restClient, restContext, err := initKafkaRest(c.AuthenticatedCLICommand, cmd)
+	if err != nil {
+		return err
+	}
+
+	clusterId, err := getClusterIdForRestRequests(restClient, restContext)
+	if err != nil {
+		return err
+	}
+
+	consumerGroup, resp, err := restClient.ConsumerGroupV3Api.ListKafkaConsumerGroups(restContext, clusterId)
+	if err != nil {
+		return kafkarest.NewError(restClient.GetConfig().BasePath, err, resp)
+	}
+
+	list := output.NewList(cmd)
+	for _, group := range consumerGroup.Data {
+		list.Add(&consumerGroupOut{
+			ClusterId:         group.ClusterId,
+			ConsumerGroupId:   group.ConsumerGroupId,
+			Coordinator:       getStringBrokerOnPrem(group.Coordinator),
+			IsSimple:          group.IsSimple,
+			PartitionAssignor: group.PartitionAssignor,
+			State:             group.State,
+		})
```

Comment:
Not sure if this is considered idiomatic Go or not, but since a lot of CLI code in general has often has these large `[resource]Out` struct literals in the implementation of `describe, update, list` commands, wouldn't it be more readable to have like "constructors" for these `[resource]Out` structs we have?

I guess we do an extra pointer copy from that approach? What do u think?